### ISSUE TITLE
Auto-Remove / avoid trailing slashes in `dspace.server.url` (possibly by treating as URL instead of String)

### ISSUE THREAD

**Issue by mwoodiupui** [DSpace/DSpace#3282]
Trivial details of e.g. `dspace.server.url` have vexing consequences.  For example:  ending the value with a slash results in a mysterious "Invalid CSRF Token" log message and failure of login.  This seems to have happened because REST was returning URLs like `https://hostname/path/to//somewhere` and token validation was unable to cope with the doubled slash.

The code contains a number of instances of treating URLs as simple strings and blindly concatenating additional path material.  There may be corresponding places where paths are compared as simple strings.

**Steps to reproduce the behavior**:
1. Make certain that the value of `dspace.server.url` ends with '/'.
2. Try to log in.

**Expected behavior**
When the code builds or examines URLs, it should treat them as structured values so that trivial oddities like "path/to//somewhere" are not created and do not cause failures.

I suggest two ways to address this:
1. Add comments to the configuration files to warn of the need to avoid problematic forms such as trailing slashes.
2. Find places where URLs are treated as unstructured strings and replace them with more robust methods of construction and analysis.  When this has been done, the warnings may be removed.  It may be useful to introduce a utility class to handle path augmentation and comparison.

**Comment by tdonohue** [comment]
Agreed with all this.  That said, I think the larger task (code cleanup) will likely need to wait for 7.1 or 7.2, as it looks to affect many areas of the codebase (and it may also affect where `dspace.ui.url` is used alongside `dspace.server.url`)

In the meantime, I've updated the docs to warn about trailing slashes & we can simply update the configs to do the same.  Thanks for finding & logging this, @mwoodiupui

**Comment by tdonohue** [comment]
Note: Step 1 (adding comments to configs) was added in #3283.  This ticket will be left open though until we improve the overall code to avoid sensitivity to trailing slashes.
